src/core/gltf2exporter/dracocompressor_p.cpp

```diff
@@ -0,0 +1,312 @@
+};
+
+template<>
+struct attribute_type_trait<double> {
+    static constexpr draco::DataType type = draco::DataType::DT_FLOAT64;
+};
+
+template<typename T>
+QByteArray packedDataInBuffer(const QByteArray &inputBuffer,
+                              int vertexSize,
+                              int offset,
+                              int stride,
+                              int count)
+{
+    QByteArray outputBuffer;
+    outputBuffer.resize(count * vertexSize * sizeof(T));
+
+    int byteMarkerPosition = offset;
+    stride = (stride == 0) ? vertexSize * sizeof(T) : stride;
+
+    for (int i = 0; i < count; ++i) {
+        std::memcpy(&(outputBuffer.data()[i * vertexSize * sizeof(T)]), &(inputBuffer.data()[byteMarkerPosition]), vertexSize * sizeof(T));
+        byteMarkerPosition += stride;
+    }
+/*
+    T *typedData = reinterpret_cast<T *>(outputBuffer.data());
+    for (int i = 0; i < vertexSize * count; ++i) {
+        qDebug() << typedData[i];
+    }
+*/
+    return outputBuffer;
+}
+
+template<typename T>
+std::pair<QString, int> addAttributeToMesh(const Qt3DRender::QAttribute &attribute, draco::Mesh &mesh)
+{
+    const auto dracoAttributeType = attributeTypeFromName(attribute.name());
+    const auto dracoDataType = attribute_type_trait<T>::type;
+
+    if (dracoAttributeType == draco::GeometryAttribute::Type::POSITION)
+        mesh.set_num_points(attribute.count());
+
+    const auto attributeBuffer = attribute.buffer();
+
+    const QByteArray packedData = packedDataInBuffer<T>(attributeBuffer->data(),
+                                                        attribute.vertexSize(),
+                                                        attribute.byteOffset(),
+                                                        attribute.byteStride(),
+                                                        attribute.count());
+
+    draco::PointAttribute meshAttribute;
+    meshAttribute.Init(attributeTypeFromName(attribute.name()),
+                       nullptr,
+                       attribute.vertexSize(),
+                       dracoDataType,
+                       false,
+                       attribute.vertexSize() * sizeof(T),
+                       0);
+    auto attId = mesh.AddAttribute(meshAttribute, true, attribute.count());
+    if (attId == -1)
+        return { {}, -1 };
+
+    for (unsigned int i = 0; i < attribute.count(); ++i)
+        mesh.attribute(attId)->SetAttributeValue(draco::AttributeValueIndex(i), &packedData.data()[i * attribute.vertexSize() * sizeof(T)]);
+
+    return { attribute.property("semanticName").toString(), attId };
+}
+
+template<typename T>
+QVector<T> createIndicesVectorFromAttribute(const Qt3DRender::QAttribute &attribute)
+{
+    const auto byteStride = attribute.byteStride();
+    const auto byteOffset = attribute.byteOffset();
+    const auto data = attribute.buffer()->data().mid(byteOffset);
+
+    QVector<T> indicesVector;
+    indicesVector.resize(attribute.count());
```

Comment:
could be written as QVector<T> indicesVector(attribute.count()); directly

Reply:
done (also is there a way to do it without initializing values ? considering that they will be set just afterwards)

Reply:
I guess you could do QVector<T> indicesVector; indicesVector.reserve(attribute.count()); which should preallocate without initializing value. Then you need to use push_back to insert values though.

Reply:
hmm I think that it would be a bit less performant - let's keep it as ` QVector<T> indicesVector(static_cast<int>(attrCount))` :)